From df4b268f63705588e29defef7d76c69afeadae52 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:35:21 -0400
Subject: [block] blk-mq: kill preempt disable_enable in blk_mq_work_fn()

Message-id: <1402670238-13916-34-git-send-email-snitzer@redhat.com>
Patchwork-id: 83834
O-Subject: [RHEL7.1 PATCH 033/150] blk-mq: kill preempt disable_enable in blk_mq_work_fn()
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 6700a678c02e47b6d50c51da2a46ff80efedb8c7
Author: Jens Axboe <axboe@fb.com>
Date:   Wed Apr 16 10:38:35 2014 -0600

    blk-mq: kill preempt disable/enable in blk_mq_work_fn()

    blk_mq_work_fn() is always invoked off the bounded workqueues,
    so it can happily preempt among the queues in that set without
    causing any issues for blk-mq.

    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq.c b/block/blk-mq.c
index 1fa01ff..b59a8d0 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -714,9 +714,7 @@ static void blk_mq_work_fn(struct work_struct *work)
 
  hctx = container_of(work, struct blk_mq_hw_ctx, delayed_work.work);
 
- preempt_disable();
  __blk_mq_run_hw_queue(hctx);
- preempt_enable();
 }
 
 static void __blk_mq_insert_request(struct blk_mq_hw_ctx *hctx,
-- 
1.7.1